src/extension/tags.c

```diff
@@ -536,13 +554,24 @@ static void _dd_request_headers(zend_array *meta_ht, zval *_server)
         memcpy(tag_p, ZSTR_VAL(key) + LSTRLEN("HTTP_"), header_name_len);
         tag_p[header_name_len] = '\0';
         dd_string_normalize_header(tag_p, header_name_len);
-        if (!zend_hash_str_exists(&_relevant_headers, tag_p, header_name_len)) {
+        bool relevant_header =
+            zend_hash_str_exists(&_relevant_headers, tag_p, header_name_len);
+        bool relevant_ip_header =
+            zend_hash_str_exists(&_relevant_ip_headers, tag_p, header_name_len);
```

Comment:
I don't think coupling the code for adding headers to the span and the code for searching for the IP is a good idea, while the duplication has a slight performance hit, it's still a better solution for many reasons, but to name a few:
- Avoid global state across tagging functions (i.e. `multiple_headers_detected`), this could be bug prone if someone reorders the functions without noticing in the future.
- Allow us to compute the IP in RINIT, as we will need to do so in order to pass it to the helper for blocking.
- The single responsibility principle applies quite nicely to functions as well.

Reply:
Would you mind to expand on this? In order to extract the IP you can use the function `dd_ip_extraction_find` outside of this module.  This code is just extracting tags from the headers `_dd_request_headers`

Reply:
From my perspective, we should have a single entry point for extracting the IP which follows the IP extraction algorithm defined in the RFC, which also includes checking if there are multiple IP headers. If I wanted to extract the IP with `dd_ip_extraction_find`, it would not take into consideration if there were multiple IP headers because this happens somewhere else entirely.

Reply:
ok

Reply:
Done! Can you re-review plz 🙏 ?